tests/nnstreamer_filter_llamacpp/unittest_filter_llamacpp.cc

```diff
@@ -0,0 +1,371 @@
+  EXPECT_EQ (new_sample_count, 1);
+}
+
+/**
+ * @brief Test case for tensor_filter llama-cpp plugin with multipleInputsMultipleOutputsAsync_p
+ */
+TEST_F (NNStreamerFilterLlamaCppTest, multipleInputsMultipleOutputsAsync_p)
+{
+  gboolean invoke_async = TRUE;
+  gint num_predict = 10;
+  if (skip_test) {
+    GTEST_SKIP ()
+        << "Skipping multipleInputsMultipleOutputsAsync_p test due to missing model file"
+           "Please download model file from https://huggingface.co/TheBloke/Llama-2-7B-Chat-GGUF";
+  }
+  ASSERT_TRUE (this->loaded);
+
+  new_sample_count = 0;
+  create_pipeline (model, invoke_async, num_predict);
+  data_push ("Hello my name is");
+  data_push ("What is AI?");
+  EXPECT_EQ (setPipelineStateSync (pipeline, GST_STATE_PLAYING, UNITTEST_STATECHANGE_TIMEOUT), 0);
+  g_usleep (3000000);
+
+  EXPECT_EQ (setPipelineStateSync (pipeline, GST_STATE_NULL, UNITTEST_STATECHANGE_TIMEOUT), 0);
+  EXPECT_GT (new_sample_count, 20);
+}
+
+/**
+ * @brief Test case for tensor_filter llama-cpp plugin with multipleInputsSingleOutputSync_p
+ */
+TEST_F (NNStreamerFilterLlamaCppTest, multipleInputsSingleOutputSync_p)
+{
+  gboolean invoke_async = FALSE;
+  gint num_predict = 10;
+  if (skip_test) {
+    GTEST_SKIP ()
+        << "Skipping multipleInputsSingleOutputSync_p test due to missing model file"
+           "Please download model file from https://huggingface.co/TheBloke/Llama-2-7B-Chat-GGUF";
+  }
+  ASSERT_TRUE (this->loaded);
+
+  new_sample_count = 0;
+  create_pipeline (model, invoke_async, num_predict);
+  data_push ("Hello my name is");
+  data_push ("What is AI?");
+  EXPECT_EQ (setPipelineStateSync (pipeline, GST_STATE_PLAYING, UNITTEST_STATECHANGE_TIMEOUT), 0);
+  g_usleep (3000000);
+
+  EXPECT_EQ (setPipelineStateSync (pipeline, GST_STATE_NULL, UNITTEST_STATECHANGE_TIMEOUT), 0);
+  EXPECT_EQ (new_sample_count, 2);
+}
+
+/**
+ * @brief Test case for tensor_filter llama-cpp plugin with invalidNumPredict_n
+ */
+TEST_F (NNStreamerFilterLlamaCppTest, invalidNumPredict_n)
+{
+  gboolean invoke_async = TRUE;
+  gint num_predict = 0;
+  if (skip_test) {
+    GTEST_SKIP ()
+        << "Skipping invalidNumPredict_n test due to missing model file"
+           "Please download model file from https://huggingface.co/TheBloke/Llama-2-7B-Chat-GGUF";
+  }
+  ASSERT_TRUE (this->loaded);
+
+  new_sample_count = 0;
+  create_pipeline (model, invoke_async, num_predict);
+  data_push ("Hello my name is");
+  EXPECT_EQ (setPipelineStateSync (pipeline, GST_STATE_PLAYING, UNITTEST_STATECHANGE_TIMEOUT), 0);
+  g_usleep (3000000);
+
+  EXPECT_EQ (setPipelineStateSync (pipeline, GST_STATE_NULL, UNITTEST_STATECHANGE_TIMEOUT), 0);
+  EXPECT_EQ (new_sample_count, 6);
+}
+
+/**
+ * @brief Test case for tensor_filter llama-cpp plugin with invalidModel_n
+ */
+TEST_F (NNStreamerFilterLlamaCppTest, invalidModel_n)
+{
+  gboolean invoke_async = TRUE;
+  gint num_predict = 10;
+  if (skip_test) {
```

Comment:
Please refer to [this review comment](https://github.com/nnstreamer/nnstreamer/pull/4739/files#r2231343182).